Source/SwiftLintFramework/Models/LinterCache.swift

```diff
@@ -18,58 +18,67 @@ public enum LinterCacheError: Error {
 public final class LinterCache {
     private var cache: [String: Any]
     private let lock = NSLock()
+    internal lazy var fileManager: LintableFileManager = FileManager.default
 
-    public init(currentVersion: Version = .current, configurationHash: Int? = nil) {
+    public init(currentVersion: Version = .current, configurationDescription: String? = nil) {
         cache = [
-            "version": currentVersion.value,
-            "files": [:]
+            Key.version.rawValue: currentVersion.value,
+            Key.files.rawValue: [:]
         ]
-        cache["configuration_hash"] = configurationHash
+        cache[LinterCache.Key.configuration.rawValue] = configurationDescription
     }
 
-    public init(cache: Any, currentVersion: Version = .current, configurationHash: Int? = nil) throws {
+    public init(cache: Any, currentVersion: Version = .current, configurationDescription: String? = nil) throws {
         guard let dictionary = cache as? [String: Any] else {
             throw LinterCacheError.invalidFormat
         }
 
-        guard let version = dictionary["version"] as? String, version == currentVersion.value else {
+        guard dictionary[Key.version.rawValue] as? String == currentVersion.value else {
             throw LinterCacheError.differentVersion
         }
 
-        if dictionary["configuration_hash"] as? Int != configurationHash {
+        guard dictionary[Key.configuration.rawValue] as? String == configurationDescription else {
             throw LinterCacheError.differentConfiguration
         }
 
         self.cache = dictionary
     }
 
     public convenience init(contentsOf url: URL, currentVersion: Version = .current,
-                            configurationHash: Int? = nil) throws {
+                            configurationDescription: String? = nil) throws {
         let data = try Data(contentsOf: url)
         let json = try JSONSerialization.jsonObject(with: data, options: [])
         try self.init(cache: json, currentVersion: currentVersion,
-                      configurationHash: configurationHash)
+                      configurationDescription: configurationDescription)
     }
 
-    public func cache(violations: [StyleViolation], forFile file: String, fileHash: Int) {
+    public func cache(violations: [StyleViolation], forFile file: String) {
+        guard let lastModification = fileManager.modificationDate(forFileAtPath: file) else {
+            return
+        }
+
         lock.lock()
-        var filesCache = (cache["files"] as? [String: Any]) ?? [:]
+        var filesCache = (cache[LinterCache.Key.files.rawValue] as? [String: Any]) ?? [:]
```

Comment:
I (inconsistently) omitted `LinterCache` within its body since it could be inferred. This one slipped through. Fixed in 43e4ed6.